pkg/ibmcsidriver/controller_helper.go

```diff
@@ -282,6 +320,57 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 	return volume, nil
 }
 
+// setSecurityGroupList
+func setSecurityGroupList(volume *provider.Volume, value string) {
+	securityGroupstr := strings.TrimSpace(value)
+	securityGroupList := strings.Split(securityGroupstr, ",")
+	var securityGroups []provider.SecurityGroup
+	for _, securityGroup := range securityGroupList {
+		securityGroups = append(securityGroups, provider.SecurityGroup{ID: securityGroup})
+	}
+	volume.VPCVolume.SecurityGroups = &securityGroups
+}
+
+// setISENIEnabled
+func setISENIEnabled(volume *provider.Volume, key string, value string) error {
+	var err error
+	if value != TrueStr && value != FalseStr {
+		err = fmt.Errorf("'<%v>' is invalid, value of '%s' should be [true|false]", value, key)
+	} else {
+		if value == TrueStr {
+			volume.VPCVolume.AccessControlMode = SecurityGroup
+		} else {
+			volume.VPCVolume.AccessControlMode = VPC
+		}
+	}
+
+	return err
+}
+
+// setPrimaryIPID
+func setPrimaryIPID(volume *provider.Volume, key string, value string) error {
+	var err error
+	if volume.VPCVolume.PrimaryIP == nil {
```

Comment:
Why is the if/else condition required?

Reply:
We iterate over the key value pairs whatever comes first we will set it out of PrimaryIPID and PrimaryIPAddress but the one which comes next will throw error as both are not allowed at the same time. So instead of creating new object we throw error. Other option is we can set this and then validate but that will need to initialise the PrimaryIP and then share across the flow.

Reply:
done